program-runtime/src/instruction_processor.rs

```diff
@@ -335,6 +335,13 @@ impl InstructionProcessor {
         }
     }
 
+    /// Remove a program
+    pub fn remove_program(&mut self, program_id: Pubkey) {
+        if let Some(position) = self.programs.iter().position(|(key, _)| program_id == *key) {
+            self.programs.remove(position);
+        }
```

Comment:
how about just using `self.programs.retain()`?

Reply:
Could, the list is guaranteed to not contain duplicates and the current method only visits until the program is found, rather than visiting all members in the list.

Reply:
i see.